src/tests/roc_sdp/test_sdp_parse.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <CppUTest/TestHarness.h>
+
+#include "roc_core/heap_allocator.h"
+#include "roc_sdp/session_description.h"
+
+namespace roc {
+namespace sdp {
+
+namespace {
+
+core::HeapAllocator allocator;
+
+} // namespace
+
+TEST_GROUP(sdp_parser) {};
+
+TEST(sdp_parser, str) {
+    SessionDescription session_description(allocator);
+    CHECK(parse_sdp("v=0\r\n"
+                    "o=test_origin 16914 1 IN IP4 192.168.58.15\r\n"
+                    "c=IN IP4 230.255.12.42/250\r\n"
+                    "m=audio 12345 RTP/AVP 10 11\r\n"
+                    "m=audio 6789 RTP/AVP 10\r\n"
+                    "c=IN IP4 231.255.12.42/250\r\n"
+                    "c=IN IP4 232.255.12.42/250",
+                    session_description));
+
+    STRCMP_EQUAL("test_origin 16914 IN 192.168.58.15", session_description.guid());
+
+    MediaDescription* last_media = session_description.last_media_description().get();
+
+    CHECK_EQUAL(6789, last_media->port());
+    CHECK_EQUAL(MediaType_Audio, last_media->type());
+    CHECK_EQUAL(MediaTransport_RTP_AVP, last_media->proto());
+    CHECK_EQUAL(10, last_media->default_payload_id());
```

Comment:
What would be best to retrieve the MediaDescription from the SessionDescription: should we return a reference to the core::List<MediaDescription> ? Or a getter that take a number/position in the list and return a reference to the related MediaDescription ?

Same to retrieve the ConnectionData from a MediaDescription: should we return a reference to the core::Array<ConnectionData> or a getter that take an id and return a reference to the related ConnectionData ?

Reply:
Regarding ConnectionData: I guess we should encapsulate how it is stored (array or something), since there are multiple forms how the same logical value can be represented (multiple `c=` lines or a single `c=` line with number-of-addresses field, see https://github.com/roc-project/roc/issues/200#issuecomment-612383960). So maybe in this case we should somehow abstract connection data by getters to handle both possible underlying representations transparently.

Regarding MediaDescription. A getter for retrieving MediaDescription by its position (index) would be inefficient (it'd require linear search), so we either can return a const reference to core::List or replicate its nextof() getter. TBH I don't see a big difference at this point. If you don't see problems with returning List, maybe this way is better because it requires less code.